Bottom: e1327a3843349e9f747aa65de7a3f12f0fc3ac9a
Top:    dd62c039b1c9f325667043d5559e355e08f18a2e
Author: Allen Short <ashort@mozilla.com>
Date:   2017-02-10 00:49:30 -0600

Don't execute query when changing data sources (fixes #29)


---
diff --git a/client/app/pages/queries/query.html b/client/app/pages/queries/query.html
index d6821e7f..a950b2b5 100644
--- a/client/app/pages/queries/query.html
+++ b/client/app/pages/queries/query.html
@@ -220,7 +220,7 @@
               <!-- End of Query Execution Status -->
 
               <!-- tabs and data -->
-              <div ng-if="showDataset" class="flex-fill p-relative">
+              <div ng-if="showDataset && !dataSourceChanged" class="flex-fill p-relative">
                 <div class="d-flex flex-column p-absolute static-position__mobile" style="left: 0; top: 0; right: 0; bottom: 0;">
                   <div class="p-10" ng-show="showLog">
                     <p>Log Information:</p>
diff --git a/client/app/pages/queries/view.js b/client/app/pages/queries/view.js
index 85d28547..ae52974e 100644
--- a/client/app/pages/queries/view.js
+++ b/client/app/pages/queries/view.js
@@ -136,6 +136,7 @@ function QueryViewCtrl(
     Notifications.getPermissions();
   };
 
+  $scope.dataSourceChanged = false;
   $scope.selectedVisualization = DEFAULT_VISUALIZATION;
   $scope.currentUser = currentUser;
   $scope.dataSource = {};
@@ -360,7 +361,7 @@ function QueryViewCtrl(
 
     $scope.dataSource = find($scope.dataSources, ds => ds.id === $scope.query.data_source_id);
     getSchema();
-    $scope.executeQuery();
+    $scope.dataSourceChanged = true;
   };
 
   $scope.setVisualizationTab = (visualization) => {
